lib/mirage_nat_lru.ml

```diff
@@ -55,9 +57,9 @@ module Storage = struct
     let lookup t key =
       MProf.Trace.label "Mirage_nat_hashtable.lookup.read";
       let t = L.table t in
-      match L.LRU.find key !t with
+      match L.LRU.find key t with
       | None -> Lwt.return_none
-      | Some (v, t') -> t := t'; Lwt.return (Some v)
+      | Some _ as r -> L.LRU.promote key t; Lwt.return r
 
```

Comment:
This can be made simpler

```
let t = L.table t in
L.LRU.promote key t;
Lwt.return (L.LRU.find key t)
```

but at the cost of one extra lookup per item. If someone is being nasty on the network this is maybe bad, or maybe it doesn't matter.